### PR TITLE
Fix pretrain bugs.

### PR DESCRIPTION
**Pull request by moiling**
1. The type of `gold_trimaps` is float32, and the range is [0, 1].
2. When pretraining, `pred_mattes` can't use trimap(`unsure` and `foreground`) generated by T-Net.

### CODE REVIEW

File: modules/shmnet.py [hunk 1]
```diff
@@ -32,13 +32,14 @@ def forward(self, batch):
             gold_trimaps = batch['trimap'].to(self.args.device)
 
             # Convert to BUF (BG, UNC, FG) classes.
-            gold_trimaps = gold_trimaps.repeat(1, 3, 1, 1)
+            gold_trimaps = gold_trimaps.repeat(1, 3, 1, 1) * 255
```
Comment (nikhilweee) [inline modules/shmnet.py:35]: `gold_trimaps` are derived from converting a trimap PNG to grayscale using PIL's `convert` function. 
https://github.com/nikhilweee/semantic-human-matting/blob/a0f1b1ff65cb8fbd1d78e25fc7da7d223ef97cf7/dataset.py#L71 
The [PIL guides](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#concept-modes) suggest that _Each pixel uses the full range of the bit depth. So a 1-bit pixel has a range of 0-1, an 8-bit pixel has a range of 0-255 and so on._ So to the best of my knowledge, I believe that trimaps are already in the range 0-255. Please correct me if I'm wrong.

File: modules/shmnet.py [hunk 2]
```diff
@@ -32,13 +32,14 @@ def forward(self, batch):
...
             gold_trimaps[:, 0, :, :] = gold_trimaps[:, 0, :, :].eq(0).float()
             gold_trimaps[:, 1, :, :] = gold_trimaps[:, 1, :, :].eq(128).float()
             gold_trimaps[:, 2, :, :] = gold_trimaps[:, 2, :, :].eq(255).float()
 
             if self.args.mode != 'end_to_end':
                 # Teacher forcing
+                _, unsure, foreground = torch.split(gold_trimaps, 1, dim=1)
```
Comment (nikhilweee) [inline modules/shmnet.py:42]: `pred_mattes` will be valid only when `unsure` and `foreground` are continuous values.
https://github.com/nikhilweee/semantic-human-matting/blob/a0f1b1ff65cb8fbd1d78e25fc7da7d223ef97cf7/modules/shmnet.py#L49
This change makes them discrete, which I believe might be a problem.

**Comment by nikhilweee** [comment]
Hey @moiling, thank you very much for your PR. I've added some comments. Please feel free to add your thoughts as well.

File: modules/shmnet.py (hunk 1 above)
Comment (moiling) [reply modules/shmnet.py:35]: Yes, you are right, but after `F.to_tensor` it will change to 0-1.
https://github.com/pytorch/vision/blob/4d3a309379f2df748922a610c5dd648ae7248bae/torchvision/transforms/functional.py#L138

File: modules/shmnet.py (hunk 2 above)
Comment (moiling) [reply modules/shmnet.py:42]: If the output of T-Net is used for fusion, the weight of T-Net will be updated when M-Net is trained separately. 
I think there should be no problem. Just like trimap-based matting algorithm, it only uses the result of unsure region, the known region is covered by mask.
Same as:
```python
pred_mattes[trimap == 0] = 0
pred_mattes[trimap == 255] = 1
```

File: modules/shmnet.py (hunk 1 above)
Comment (nikhilweee) [reply modules/shmnet.py:35]: Thank you! I wasn't aware of this fact. Appreciate the help.

File: modules/shmnet.py (hunk 2 above)
Comment (nikhilweee) [reply modules/shmnet.py:42]: Fair enough. I'll take that. Thanks again!

**Comment by nikhilweee** [comment]
Thanks @moiling! Appreciate it. 😄
